src/czar/CzarChunkMap.h

```diff
@@ -0,0 +1,149 @@
+// -*- LSST-C++ -*-
+/*
+ * LSST Data Management System
+ *
+ * This product includes software developed by the
+ * LSST Project (http://www.lsst.org/).
+ *
+ * This program is free software: you can redistribute it and/or modify
+ * it under the terms of the GNU General Public License as published by
+ * the Free Software Foundation, either version 3 of the License, or
+ * (at your option) any later version.
+ *
+ * This program is distributed in the hope that it will be useful,
+ * but WITHOUT ANY WARRANTY; without even the implied warranty of
+ * MERCHANTABILITY or FITNESS FOR A PARTICULAR PURPOSE.  See the
+ * GNU General Public License for more details.
+ *
+ * You should have received a copy of the LSST License Statement and
+ * the GNU General Public License along with this program.  If not,
+ * see <http://www.lsstcorp.org/LegalNotices/>.
+ */
+
+#ifndef LSST_QSERV_CZAR_CZARCHUNKMAP_H
+#define LSST_QSERV_CZAR_CZARCHUNKMAP_H
+
+// System headers
+#include <chrono>
+#include <cstdint>
+#include <memory>
+#include <mutex>
+#include <set>
+#include <string>
+#include <sstream>
+
+// Third party headers
+#include <nlohmann/json.hpp>
+
+// Qserv headers
+#include "util/Issue.h"
+
+namespace lsst::qserv::qmeta {
+class QMeta;
+}
+
+namespace lsst::qserv::czar {
+
+class ChunkMapException : public util::Issue {
+public:
+    ChunkMapException(Context const& ctx, std::string const& msg) : util::Issue(ctx, msg) {}
+};
+
+class CzarChunkMap {
+public:
+    using Ptr = std::shared_ptr<CzarChunkMap>;
+
+    CzarChunkMap() = delete;
+
+    static Ptr create(std::shared_ptr<qmeta::QMeta> const& qmeta) {
+        return Ptr(new CzarChunkMap(qmeta));
+    }
+
+    class WorkerChunksData;
+
+    class ChunkData {
+    public:
+        using Ptr = std::shared_ptr<ChunkData>;
+        ChunkData(int chunkId_) : _chunkId(chunkId_) {}
+        int64_t const _chunkId;
+        int64_t _totalBytes = 0;
+        std::weak_ptr<WorkerChunksData> _primaryScanWorker;
+
+        /// &&& doc
+        void calcTotalBytes();
+
+        /// &&& doc
+        void addToWorkerHasThis(std::shared_ptr<WorkerChunksData> const& worker);
+
+        /// Key is databaseName+tableName, value is size in bytes.
+        std::map<std::pair<std::string, std::string>, int64_t> _dbTableMap;
+
+        /// Map of workers that have this chunk
+        std::map<std::string, std::weak_ptr<WorkerChunksData>> _workerHasThisMap;
+
+        std::string dump() const;
+    };
+
+    class WorkerChunksData {
+    public:
+        using Ptr = std::shared_ptr<WorkerChunksData>;
+        WorkerChunksData(std::string const& wId) : _workerId(wId) {}
```

Comment:
This isn't very important. However, it's always easier to read a code where there is a direct mapping of the construction parameters to the corresponding members like in:
```
 WorkerChunksData(std::string const& workerId) : _workerId(workerId) {}
```
...unless you're going to have the public/protected method like:
```
std::string const& workerId() const { return _workerId; }
```
In the latter case, the following usually works well:
```
WorkerChunksData(std::string const& workerId_) : _workerId(workerId_) {}
```

Reply:
Changed name to workerId.